src/maybe/maybe.py

```diff
@@ -81,7 +89,7 @@ def unwrap(self) -> T:
         """
         return self._value
 
-    def unwrap_or(self, _default: U) -> T:
+    def unwrap_or(self, _default: object) -> T:
```

Comment:
Yep, gotcha. Sort of a catch-22 situation because using a type var that's "unbound" causes its own type warning, but I think the lesser of the 2 problems is the "unbound" (invalid use) type warning, which we can safely ignore.  I think, as you point out, that having the type error on the other end (when failing to pass a function to `Some.ok_or_else`) is more important.